File: translated/f-to-c/DRB098-simd2-orig-no.c
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.

SPDX-License-Identifier: (BSD-3-Clause)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//Two-dimension array computation with a vetorization directive
//collapse(2) makes simd associate with 2 loops.
//Loop iteration variables should be predetermined as lastprivate. No data race pairs.

#include <omp.h>
#include <stdio.h>
#include <stdlib.h>

int main()
{
    double** a;
    double** b;
    double** c;
    int len, i, j;

    len = 100;

    // Allocate 2D arrays
    a = (double**)malloc(len * sizeof(double*));
    b = (double**)malloc(len * sizeof(double*));
    c = (double**)malloc(len * sizeof(double*));
    for (i = 0; i < len; i++) {
        a[i] = (double*)malloc(len * sizeof(double));
        b[i] = (double*)malloc(len * sizeof(double));
        c[i] = (double*)malloc(len * sizeof(double));
    }

    for (i = 0; i < len; i++) {
        for (j = 0; j < len; j++) {
            a[i][j] = (double)(i + 1) / 2.0;  // Adjust for 0-based indexing
            b[i][j] = (double)(i + 1) / 3.0;
            c[i][j] = (double)(i + 1) / 7.0;
        }
    }

    #pragma omp simd collapse(2)
    for (i = 0; i < len; i++) {
        for (j = 0; j < len; j++) {
            c[i][j] = a[i][j] * b[i][j];
        }
    }

    printf("c(50,50) = %f\n", c[49][49]);  // Adjust for 0-based indexing

    // Free 2D arrays
    for (i = 0; i < len; i++) {
        free(a[i]);
        free(b[i]);
        free(c[i]);
    }
    free(a);
    free(b);
    free(c);

    return 0;
} 

File: translated/f-to-go-automodel-bad/DRB098-simd2-orig-no.go
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
// and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.
//
// SPDX-License-Identifier: (BSD-3-Clause)
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

//Two-dimension array computation with a vetorization directive
//collapse(2) makes simd associate with 2 loops.
//Loop iteration variables should be predetermined as lastprivate. No data race pairs.

package main

import (
	"fmt"
)

func main() {
	var a, b, c [][]float64
	var length int

	length = 100
	a = make([][]float64, length)
	b = make([][]float64, length)
	c = make([][]float64, length)

	for i := range a {
		a[i] = make([]float64, length)
		b[i] = make([]float64, length)
		c[i] = make([]float64, length)
	}

	for i := 1; i <= length; i++ {
		for j := 1; j <= length; j++ {
			a[i-1][j-1] = float64(i) / 2.0
			b[i-1][j-1] = float64(i) / 3.0
			c[i-1][j-1] = float64(i) / 7.0
		}
	}

	//$omp simd collapse(2)
	for i := 1; i <= length; i++ {
		for j := 1; j <= length; j++ {
			c[i-1][j-1] = a[i-1][j-1] * b[i-1][j-1]
		}
	}
	//$omp end simd

	fmt.Printf("c(50,50) = %f\n", c[49][49])
}
